examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -0,0 +1,235 @@
+
+    // MicrowaveOvenMode types
+    using ModeTagStructType              = detail::Structs::ModeTagStruct::Type;
+    ModeTagStructType modeTagsNormal[1]  = { { .value = to_underlying(MicrowaveOvenMode::ModeTag::kNormal) } };
+    ModeTagStructType modeTagsDefrost[1] = { { .value = to_underlying(MicrowaveOvenMode::ModeTag::kDefrost) } };
+
+    const detail::Structs::ModeOptionStruct::Type kModeOptions[2] = {
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Normal"),
+                                                 .mode     = ModeNormal,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(modeTagsNormal) },
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Defrost"),
+                                                 .mode     = ModeDefrost,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(modeTagsDefrost) }
+    };
+
+    // Operational States
+    const OperationalState::GenericOperationalState opStateList[4] = {
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kStopped)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kPaused)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kError)),
+    };
+
+    app::DataModel::List<const OperationalState::GenericOperationalState> mOperationalStateList =
+        Span<const OperationalState::GenericOperationalState>(opStateList);
+
+    const OperationalState::GenericOperationalPhase opPhaseList[1] = {
+        // Phase List is null
+        OperationalState::GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
+    };
+
+    Span<const OperationalState::GenericOperationalPhase> mOperationalPhaseList =
+        Span<const OperationalState::GenericOperationalPhase>(opPhaseList);
+
+public:
+    /**
+     * This class is responsible for initialising all the microwave oven device clusters and managing the interactions between them
+     * as required by the specific "business logic". See the state machine diagram.
+     * @param aClustersEndpoint The endpoint ID where all the microwave oven clusters exist.
+     */
+    explicit ExampleMicrowaveOvenDevice(EndpointId aClustersEndpoint) :
+        mMicrowaveOvenControlInstance(this, aClustersEndpoint, MicrowaveOvenControl::Id, &mOperationalStateInstance,
+                                      &mMicrowaveOvenModeInstance),
+        mOperationalStateInstance(this, aClustersEndpoint, OperationalState::Id),
+        mMicrowaveOvenModeInstance(this, aClustersEndpoint, MicrowaveOvenMode::Id,
+                                   chip::to_underlying(MicrowaveOvenMode::Feature::kOnOff))
```

Comment:
Microwave oven mode doesn't support StartUpM
ode and OnMode, please disable DEFONOFF feature.